core/common/src/geometry/GeodeticDatum.ts

```diff
@@ -0,0 +1,559 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+// cspell:ignore JSONXYZ, ETRF, OSGB, DHDN
+
+import { Vector3d, XYAndZ } from "@bentley/geometry-core";
+import { GeodeticEllipsoid, GeodeticEllipsoidProps } from "./GeodeticEllipsoid";
+
+/** Holds 3 components of a Positional Vector rotation definition in arc seconds
+ *  @alpha
+ */
+export interface XyzRotationProps {
+  /** X rotation component in arc second */
+  x: number;
+  /** Y rotation component in arc second*/
+  y: number;
+  /** Z rotation component in arc second*/
+  z: number;
+}
+
+/** Hold 3 components data of a Positional Vector rotation definition in arc seconds
+ *  @alpha
+ */
+export class XyzRotation implements XyzRotationProps {
+  /** X rotation component in arc second */
+  public x!: number;
+  /** Y rotation component in arc second*/
+  public y!: number;
+  /** Z rotation component in arc second*/
+  public z!: number;
+
+  public constructor(data?: XyzRotationProps) {
+    this.initialize(data);
+  }
+
+  /** @internal */
+  public initialize(_data?: XyzRotationProps) {
+    if (_data) {
+      this.x = _data.x;
+      this.y = _data.y;
+      this.z = _data.z;
+    }
+  }
+
+  /** @internal */
+  public static fromJSON(data: XyzRotationProps): XyzRotation {
+    return new XyzRotation(data);
+  }
+
+  /** @internal */
+  public toJSON(): XyzRotationProps {
+    return { x: this.x, y: this.y, z: this.z };
+  }
+
+  /** @internal */
+  public equals(other: XyzRotation): boolean {
+    return (this.x === other.x && this.y === other.y && this.z === other.z);
+  }
+}
+
+/** Enum indicating the geodetic transformation method
+ *  @alpha
+ */
+export enum GeodeticTransformMethod {
```

Comment:
Consider using a union type `"None" | "Geocentric" | ...` instead...compiler can reason better about such types and they are less verbose.

Reply:
Done